File: llvm-gcc-4.2-2.9/gcc/testsuite/gcc.apple/block-weak.c
/* APPLE LOCAL file blocks 7184496 */
/* Test __weak attribute on __block objects. */
/* { dg-do compile { target *-*-darwin* } } */
/* { dg-options "-fblocks -ObjC -fobjc-gc" { target *-*-darwin* } } */
/* Radar 7184496 */

void nonweakfofo(void) {
  __block (^weakSelf)();
}

void weakfofo(void) {
  __block __weak (^weakSelf)();
}

/* { dg-final { scan-assembler "151" } } */
/* { dg-final { scan-assembler "135" } } */

File: llvm-gcc-4.2-2.9/gcc/testsuite/objc.dg/block-weak.m
/* APPLE LOCAL file __block copy/dispose 7284529 */
/* { dg-do compile { target *-*-darwin* } } */
/* { dg-options "-fblocks" } */

@interface ShortcutsController
@end

@class barbar;
@implementation ShortcutsController
- (void) _markConflicts
{
 __attribute__((__blocks__(byref))) barbar * matchDict;
 __attribute__((__blocks__(byref))) void (^markConflictsBlock)();
}
@end
